pkgs/applications/networking/cluster/fluxcd/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, buildGoModule, fetchFromGitHub, installShellFiles }:
+
+buildGoModule rec {
+  pname = "fluxcd";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "fluxcd";
+    repo = "flux2";
+    rev = "v${version}";
+    sha256 = "125im8br7x8djd6zagvikpc02k55pxbd97rjj3g2frj9plbryh8n";
+  };
+
+  vendorSha256 = "0f818a0z71nl061db93iqb87njx66vbrra1zh92warbx8djdsr7k";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  doCheck = false;
+
+  subPackages = [ "cmd/flux" ];
+
+  buildFlagsArray = [ "-ldflags=-s -w -X main.VERSION=${version}" ];
```

Comment:
I'm not super familiar with the build toolchain, why is this necessary?

Reply:
at least the `-ldflags=-s -w` bit, the version bit makes sense

Reply:
Thanks for reviewing! 
Both options are necessary, to know more about the go toolchain, you can take a look here: https://nixos.org/manual/nixpkgs/stable/#ssec-go-modules. It explains the `subPackages` option.

The `buildFlagsArray` values are taken from upstream: 

https://github.com/fluxcd/flux2/blob/0ba6fc1b366bd6f2ce231d3ac8a95eac57ff8554/.goreleaser.yml#L6

The `-X main.VERSION` is needed because without it, the `flux --version` command will return version 0

Those two flags are pretty common between the go derivations

Reply:
Thanks for the explanation!!

Reply:
Ideally the link to the upstream source would be mentioned in a comment to avoid future confusion. The rest of the diff looks good though, and since its a very common flag this is not very important. Just mentioning this for future PRs. If something needs explanation in the review, its usually best to explain it in a comment. [This](https://mtlynch.io/code-review-love/#4-answer-questions-with-the-code-itself) blog post argues the point very well.